src/java/picard/filter/CountingFilter.java

```diff
@@ -0,0 +1,39 @@
+package picard.filter;
+
+import htsjdk.samtools.AlignmentBlock;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.filter.SamRecordFilter;
+
+/**
+ * A SamRecordFilter that counts the number of aligned bases in the reads which it filters out. Abstract and designed
+ * to be subclassed to implement the desired filter.
+ */
+public abstract class CountingFilter implements SamRecordFilter {
+    private long filteredRecords = 0;
+    private long filteredBases = 0;
+
+    /** Gets the number of records that have been filtered out thus far. */
+    public long getFilteredRecords() { return this.filteredRecords; }
+
+    /** Gets the number of bases that have been filtered out thus far. */
+    public long getFilteredBases() { return this.filteredBases; }
+
+    @Override
+    public final boolean filterOut(final SAMRecord record) {
+        final boolean filteredOut = reallyFilterOut(record);
+        if (filteredOut) {
+            ++filteredRecords;
+            for (final AlignmentBlock block : record.getAlignmentBlocks()) {
+                this.filteredBases += block.getLength();
+            }
+        }
+        return filteredOut;
+    }
+
+    abstract public boolean reallyFilterOut(final SAMRecord record);
```

Comment:
With this being hoisted out into picard.filters, it would be nice to add a little more javadoc, especially to this method.